daemon/openrazer_daemon/hardware/device_base.py

```diff
@@ -104,13 +134,176 @@ def __init__(self, device_path, device_number, config, testing=False, additional
+            },
+
+            "scroll": {
+                ('razer.device.misc', 'getCurrentScrollEffect', self.get_current_scroll_effect, None, 's'),
+                ('razer.device.misc', 'getCurrentScrollEffectColors', self.get_current_scroll_effect_colors, None, 'ay'),
+                ('razer.device.misc', 'getCurrentScrollEffectSpeed', self.get_current_scroll_effect_speed, None, 'i'),
+                ('razer.device.misc', 'getCurrentScrollWaveDir', self.get_current_scroll_wave_dir, None, 'i'),
+            },
+
+            "left": {
+                ('razer.device.misc', 'getCurrentLeftEffect', self.get_current_left_effect, None, 's'),
+                ('razer.device.misc', 'getCurrentLeftEffectColors', self.get_current_left_effect_colors, None, 'ay'),
+                ('razer.device.misc', 'getCurrentLeftEffectSpeed', self.get_current_left_effect_speed, None, 'i'),
+                ('razer.device.misc', 'getCurrentLeftWaveDir', self.get_current_left_wave_dir, None, 'i'),
+            },
+
+            "right": {
+                ('razer.device.misc', 'getCurrentRightEffect', self.get_current_right_effect, None, 's'),
+                ('razer.device.misc', 'getCurrentRightEffectColors', self.get_current_right_effect_colors, None, 'ay'),
+                ('razer.device.misc', 'getCurrentRightEffectSpeed', self.get_current_right_effect_speed, None, 'i'),
+                ('razer.device.misc', 'getCurrentRightWaveDir', self.get_current_right_wave_dir, None, 'i'),
+            }
+        }
+
         for m in methods:
             self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
             self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
 
+        if 'set_static_effect' in self.METHODS:
+            self.zone["backlight"]["present"] = True
+            for m in effect_methods["backlight"]:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
+        for i in self.ZONES[1:]:
+            if 'set_' + i + '_static' in self.METHODS or 'set_' + i + '_static_naga_hex_v2' in self.METHODS or '`set_' + i + 'active' in self.METHODS:
+                self.zone[i]["present"] = True
+                for m in effect_methods[i]:
+                    self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                    self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
         # Load additional DBus methods
         self.load_methods()
 
+        # load last effects
+        for i in self.ZONES:
+            if self.zone[i]["present"]:
+                if self.config.has_section(self.storage_name):
+                    try:
+                        self.zone[i]["effect"] = self.config[self.storage_name][i + '_effect']
+                    except KeyError:
+                        self.zone[i]["effect"] = 'spectrum'
```

Comment:
Yep, sorry about that. Fixed.

Reply:
However, I didn't remove the re-initialization code in the `except ValueError` because it handles incomplete color reading (instead of leaving us with a broken color set, it resets the colors).